dipy/workflows/base.py

```diff
@@ -0,0 +1,124 @@
+
+    def __init__(self, prog=None, usage=None, description=None, epilog=None,
+                 version=None, parents=[],
+                 formatter_class=arg.RawTextHelpFormatter,
+                 prefix_chars='-', fromfile_prefix_chars=None,
+                 argument_default=None, conflict_handler='resolve',
+                 add_help=True):
+
+        """ Augmenting the argument parser to allow automatic creation of
+        arguments from workflows
+
+        Parameters
+        -----------
+        prog : None
+            The name of the program (default: sys.argv[0])
+        usage : None
+            A usage message (default: auto-generated from arguments)
+        description : str
+            A description of what the program does
+        epilog : str
+            Text following the argument descriptions
+        version : None
+            Add a -v/--version option with the given version string
+        parents : list
+            Parsers whose arguments should be copied into this one
+        formatter_class : obj
+            HelpFormatter class for printing help messages
+        prefix_chars : str
+            Characters that prefix optional arguments
+        fromfile_prefix_chars : None
+            Characters that prefix files containing additional arguments
+        argument_default : None
+            The default value for all arguments
+        conflict_handler : str
+            String indicating how to handle conflicts
+        add_help : bool
+            Add a -h/-help option
+        """
+
+        iap = IntrospectiveArgumentParser
+        super(iap, self).__init__(prog, usage, description, epilog, version,
+                                  parents, formatter_class, prefix_chars,
+                                  fromfile_prefix_chars, argument_default,
+                                  conflict_handler, add_help)
+
+        self.doc = None
+
+    def add_workflow(self, workflow):
+        specs = inspect.getargspec(workflow)
+        doc = inspect.getdoc(workflow)
+
+        if has_ndoc:
+            self.doc = ndoc.docscrape.NumpyDocString(doc)['Parameters']
+
+        args = specs.args
+        defaults = specs.defaults
+
+        len_args = len(args)
+        len_defaults = len(defaults)
+
+        # Arguments with no defaults (Positional)
+        cnt = 0
+        for i in range(len_args - len_defaults):
+            if has_ndoc:
+                help_msg = ''.join(self.doc[i][2])
+                self.add_argument(args[i], help=help_msg)
+            else:
+                self.add_argument(args[i])
+            cnt += 1
+
+        # Arguments with defaults (Optional)
+        for i in range(cnt, len_args):
+            if has_ndoc:
+                typestr = self.doc[i][1]
+                dtype = self._select_dtype(typestr)
+                help_msg = ' '.join(self.doc[i][2])
+                self.add_argument('--' + args[i], action='store', type=dtype,
```

Comment:
There is a bug here when boolean is used. The argparser is not smart with bool but here is a suggested solution. You can specify 0 or 1. But let's talk about it.
```
                if 'bool' in typestr:
                    self.add_argument('--' + args[i], choices=[0, 1], type=int,
                                      metavar=dtype.__name__, help=help_msg)
                else:
                    self.add_argument('--' + args[i], action='store',
                                      type=dtype,
                                      metavar=dtype.__name__, help=help_msg)
```